### PR TITLE
feat: add asset list partition parameter

### PR DESCRIPTION
**Pull request by ghost**
This is a feature needed by Discovery. Currently, the JS sdk provides no easy way to fetch all assets of a tenant. However, for running Entity Matcher in discovery we would like to fetch everything quickly in parallel through partitions.

This solution simply enables the user to declare partitions. There's a more complex solution that can be useful that actually abstract out the fetching process itself.
```js
const results = await Promise.all(
      [...Array(n).keys()].map(async index => {
        let response = await sdk.assets.list({
          filter: { rootIds: [{ id: selectedRootAssetId }] },
          limit: 1000,
          // @ts-ignore
          partition: `${index + 1}/n`,
        });
        while (response.nextCursor) {
          response = await sdk.assets.list({
            filter: { rootIds: [{ id: selectedRootAssetId }] },
            limit: 1000,
            cursor: response.nextCursor,
            // @ts-ignore
            partition: `${index + 1}/n`,
          });
        }
...
)
```

### CODE REVIEW

**Comment by polomani** [comment]
Was it released? I don't see it in documentation

File: src/__tests__/resources/assets.integration.spec.ts [hunk 1]
```diff
@@ -256,4 +256,10 @@ describe('Asset integration test', () => {
       .autoPagingToArray({ limit });
     expect(result.length).toBe(limit);
   });
+
+  test('list partition', async () => {
+    const limit = 5;
+    const result = await client.assets.list({ partition: '1/10', limit });
```
Comment (polomani) [inline src/__tests__/resources/assets.integration.spec.ts:271]: does it work well with autoPaging methods?

Comment (ghost) [reply src/__tests__/resources/assets.integration.spec.ts:271]: I dont think so. Partition is only used when you dont want pagination (fetch everything ignoring limits). So i dont think we should think about that case?

**Comment by ghost** [comment]
@polomani @f1cognite @maciejmrowiec As discussed in the chat, this would enable `partition` for assets, events and timeseries.

File: src/__tests__/resources/timeseries.integration.spec.ts [hunk 2]
```diff
@@ -106,6 +106,12 @@ describe('Timeseries integration test', () => {
       .autoPagingToArray({ limit: 100 });
   });
 
+  test('list partition', async () => {
+    await client.timeseries
+      .list({ includeMetadata: false, partition: '1/10' })
+      .autoPagingToArray({ limit: 100 });
```
Comment (polomani) [inline src/__tests__/resources/timeseries.integration.spec.ts:112]: you don't test a response here?

**Review by polomani — COMMENTED** [review]
Tests are failing, did you try locally?

File: src/__tests__/resources/assets.integration.spec.ts (hunk 1 above)
Comment (BugGambit) [reply src/__tests__/resources/assets.integration.spec.ts:271]: Partition should work excellent with pagination?
Essentially you just divide the data set in the backend before fetching so still need pagination for each partition

**Comment by polomani** [comment]
@dlcognite `Unexpected field - filter.partition | code: 400` leave that on me, I can implement support for timeseries myself, I know what this issue. But the rest you can do

**Review by polomani — APPROVED** [review]

**Pull request merged**